File: 876 - Middle of the Linked List/twopointers.java

// Definition for singly-linked list.
public class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 }
 
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode start = new ListNode(0, head);
        ListNode slow = start;
        ListNode fast = start;

        while(fast != null) {
            slow = slow.next;
            fast = fast.next == null ? fast.next : fast.next.next;
        }
        return slow;
    }
}

File: 141 - Linked List Cycle/twopointers.ts
function hasCycle(head: ListNode | null): boolean {
    let slow : ListNode = head;
    let fast : ListNode = head;
    while(fast != null && fast.next != null){
        slow = slow.next;
        fast = fast.next.next;
        if( slow == fast ) {
            return true;
        }
    }
    return false;
};